rtb/analisi_dei_requisiti_ver0.3.0.typ

```diff
@@ -0,0 +1,325 @@
+  3. Il sistema verifica se le credenziali inserite sono valide o meno.
+  4. Nel caso in cui le credenziali fossero valide, l'utente o l'amministratore (a seconda del ruolo all'interno del sistema) accede alla propria _dashboard_ personale.
+- *Estensione*: Visualizzazione del messaggio di credenziali errate (UC6).
+
+=== UC6 - Visualizzazione del messaggio di errore
+- *Attori principali*: Utente non autenticato.
+- *Precondizioni*: l'utente non autenticato tenta di accedere alla _dashboard_ con delle credenziali errate.
+- *Postcondizioni*: l'utente non autenticato riceve un messaggio di errore.
+- *Trigger*: l'utente non autenticato inserisce delle credenziali errate.
+- *Scenario principale*:
+  1. L'utente inserisce delle credenziali errate durante la fase di autenticazione alla _dashboard_.
+  2. L'utente riceve un messaggio di errore, segnalando il fatto che le credenziali di accesso inserite siano invalide e invitando a riprovare l'autenticazione.
+#v(20pt)
+#figure(
+  image("../assets/use_cases/UC5-6.svg"),
+  caption: [Diagramma dei casi d'uso UC5 e UC6],
+)
+
+#pagebreak()
+
+=== UC7 - Visualizzazione della mappa dei mezzi noleggiati
+- *Attore principale*: Amministratore.
+- *Precondizioni*: l'amministratore del sistema è autenticato e ha accesso alla _dashboard_ del sistema.
+- *Postcondizioni*: l'amministratore ottiene una visione chiara della posizione e del movimento dei mezzi attualmente in uso all'interno di una mappa.
+- *Trigger*: l'amministratore intende visualizzare la posizione dei mezzi noleggiati.
+- *Scenario principale*:
+  1. L'amministratore è collegato e autenticato nella _dashboard_ del sistema.
+  2. La _dashboard_ mette a disposizione una mappa interattiva con i mezzi attualmente a noleggio, la cui posizione viene indicata attraverso dei _#rifGlossario("marker")_.
+#v(20pt)
+#figure(
+  image("../assets/use_cases/UC7.svg"),
+  caption: [Diagramma del caso d'uso UC7],
+)
+
+#pagebreak()
+
+=== UC8 - Invio del'annuncio pubblicitario generato
+- *Attore principale*: LLM.
+- *Precondizioni*: l'LLM, tramite una richiesta #rifGlossario("API"), deve aver ricevuto i _prompt_ di generazione dell'annuncio, come la profilazione dell'utente e il punto di interesse per cui si vuole generare l'annuncio.
+- *Postcondizioni*: l'LLM genera l'annuncio basandosi sulla profilazione dell'utente.
+- *Trigger*: il sistema ha fatto una richiesta di generazione tramite le API dell'LLM.
+- *Scenario principale*:
+  1. L'LLM riceve la richiesta da parte del sistema
+  2. L'LLM genera l'annuncio pubblicitario basandosi sui dati ricevuti come _prompt_.
+  3. L'LLM invia l'annuncio pubblicitario generato al sistema.
+- *Estensione*: Annuncio non generato (UC9).
+=== UC9 - Annuncio non generato
+- *Attore principale*: LLM
+- *Precondizioni*: l'LLM considera il punto di interesse non pertinente in base alla profilazione dell'utente per cui si vuole generare l'annuncio.
```

Comment:
- la pre dovrebbe essere la stessa di UC8 credo, cioè la LLM ha ricevuto il prompt (la valutazione negativa è oggetto del caso d'uso in sé quindi dello scenario)

Reply:
**Segnalibro**

Reply:
Ecco... io avrei detto diversamente, è il caso d'uso d'estensione per l'"errore"

Reply:
Direi di lasciare in sospeso come nel caso di UC3, rientra nella stessa domanda da fare a Cardin. Chiedo tuttavia il parere di @MarinDavide.

Reply:
Lascio il todo